subcommands/targets/show.go

```diff
@@ -45,19 +54,51 @@ func init() {
 	showAppCmd.Flags().Bool("manifest", false, "Show an app docker manifest")
 
 	sbomCmd := &cobra.Command{
-		Use:   "sboms <version> [<build/run>]",
+		Use:   "sboms <version> [<build/run> [<artifact>]] ",
 		Short: "Show SBOMs for a specific target.",
 		Run:   doShowSboms,
-		Args:  cobra.RangeArgs(1, 2),
+		Args:  cobra.RangeArgs(1, 3),
 		Example: `
   # Show all SBOM files for Target version 42:
   fioctl targets show sboms 42
 
   # Show a subset of the SBOMS for this target. In this case, the 32-bit Arm
   # container SBOMS:
-  fioctl targets show sboms 42 41/build-armhf`,
+  fioctl targets show sboms 42 41/build-armhf
+ 
+  # Show overview of a specific SBOM:
+  fioctl targets show sboms 42 41/build-armhf alpine:latest/arm.spdx.json
+  
+  # Show overview of a specific SBOM as CSV:
+  fioctl targets show sboms --format csv 42 41/build-armhf alpine:latest/arm.spdx.json`,
+	}
+
+	sbomFormats = append(sbomFormats, sbomFormat{
+		name:        "",
+		contentType: "application/spdx.json",
+	})
+	sbomFormats = append(sbomFormats, sbomFormat{
+		name:        "spdx",
+		contentType: "application/spdx.json",
+	})
+	sbomFormats = append(sbomFormats, sbomFormat{
+		name:        "cyclonedx",
+		contentType: "application/cyclone.json",
+	})
+	sbomFormats = append(sbomFormats, sbomFormat{
+		name:        "csv",
+		contentType: "text/csv",
+	})
```

Comment:
Can you make this a map?
I think it could even be a literal map like below:

```
var sbomFormats = map[string]string{
    "spdx": "application/spdx.json",
    ...
}

func parseSbomFormat(format string) (res sbomFormat) {
    if ct, ok := sbomFormats[format]; ok {
        res = sbomFormat{format, ct}
    } else {
        DieNotNil(fmt.Errorf("Unknown --format: %s", format))
    }
    return
}
```

That should save quite a few lines of code and make it more clear.

Reply:
Thanks. I'm finding a struct a little easier still because there's special handling for the "table" output. However, by setting a default CLI option and making this a helper function, I've got the code a lot easier:

 https://github.com/foundriesio/fioctl/pull/186/commits/70d47e90517ee5996b47a21df0e4d4de0a1d5f1d

Reply:
It is still possible to use the struct with a map just like I proposed in my example (see `res = sbomFormat{format, ct}`) or in a different way.
The whole idea was to use a map instead of a list as it is easier to work with in this case (doesn't require an append or a for loop to search).

I'm fine if you like the current approach more.

Reply:
its just gets weird because the map doesn't give you control over the order you display the format options. It's not huge, but I was trying to present them in a certain order.

Reply:
That's reasonable.

Now, what about just hard-coding a list of options for the help message?
It's just 4 simple comma-separated words; do we need all this hussle for the CLI help message?

Reply:
Note, that I usually prefer a natural language in help messages like `allowed formats: foo, bar, or baz`.
Now go write a function which does that (I know it's possible, but we'd need to support it, and add localization at 2025).

Reply:
i've rewritten as an array and hard-coded the logic:

ebc402aa36a05fdd81f602410e1ee2671f6a1bb3

Reply:
Looks much simpler now, thanks.

You've got my LGTM earlier anyway, so ship whatever way you like more.